### ISSUE TITLE
React Reader - a react wrapper for epub.js

### ISSUE THREAD

**Issue by gerhardsletten** [futurepress/epub.js#402]
Hi

I have been looking at how to make a react wrapper for epub.js, and if anyone have some suggestions or would like and want to help out. 

I am using av fork of the v0.3 version (just fixing the main-url and turn off instruments debug in book.js) but probably it will need to wait for v0.3 is realeased before it works with all books work.

Take a look at [React Reader](http://gerhardsletten.github.io/react-reader/)

**Comment by fchasen** [comment]
This is fantastic and good timing as well.

I've been updating the https://github.
com/futurepress/epub.js/tree/view_managers branch to work better with different rendereres such as react and react-native

It's still a work in progress but I'll reach out as soon as I have something to try.

**Comment by jarecsni** [comment]
Yes, I'm interested in adapting epub.js to React as well. At the moment I'm migrating the epub.js code base to ES6, let me know if you're interested in it, I'm happy to share my results. I still need a week maybe to finish this though.

**Comment by gerhardsletten** [comment]
In this react-reader I actually use the regular iframe renderer, but I have seen your react native sample, but if I understand the view_managers branch correct, it only deliveres the whole html-content of the chapter. So to use it, you would need some kind on module that render it horizontal like it does in the current version of ebub.js yourself. But does that exist now?

**Comment by jarecsni** [comment]
I'm a bit new to both react and epubjs, and I'm guessing this is more a question about react - react native than Epubjs, but still. Would a React port of Epubjs work in a ReactNative environment without any modifications? Like your react-reader, does it work in ReactNative?

**Comment by ssomnoremac** [comment]
@jarecsni, @gerhardsletten, I'm planning this week to push through some changes on v3.0, specifically as it concerns #376 , which hasn't been resolved. Have either of you become more familiar with the v3 code base to the point where you can contribute? I think if we can get this off Fred's plate he can focus some other issues to get epub.js to both React and React-Native.

**Comment by gerhardsletten** [comment]
@ssomnoremac I like to help out, but @jarecsni mention that he has spent some time to migrate to es6, so its that published somewhere? 

In my dives into the code I still think the flow within epub.js is a little complicated, the different files (modules) is communicating with each other multiple times, and with the possibility to chose different rendering methods, should the iframe render be moved  out as a seperate modules, or at leased a named export. On NPM there is also multiple modules that can extract metadata and html content from an epub-file, so should we not use one of these instead of "building the wheel" another time?

**Comment by ssomnoremac** [comment]
@gerhardsletten somehow I didn't get notified of your reply, my apologies. I haven't heard of any work to move epubjs to es6 although I'd be very interested in that effort! @jarecsni , is it true?

I'm becoming familiar with the codebase and there is a bit left to be cleaned up for 0.3 but I think the meta extracting is not the part to redo as it should be working fine. If anything, the events and ViewManagers needed to fixed up a bit. I'm adding functionality for fixed layout spreads on my side. should push it [here ](https://github.com/libraryforall/epub.js/tree/fixed_spreads)within the week, meanwhile working on docs.

**Comment by gerhardsletten** [comment]
@ssomnoremac Could you create a list of what we need to fix?

**Comment by ssomnoremac** [comment]
I've been focusing on just one area so it wouldn't be a complete list. I'll put something up soon, not on this thread.
